cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/ReplicationStatusKey.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import co.cask.cdap.api.common.Bytes;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Handling of Row keys for the Replication State Table
+ */
+public final class ReplicationStatusKey {
+
+  private final byte[] key;
+
+  public ReplicationStatusKey(byte[] prefix, byte[] regionName, byte[] regionServerID) {
+    byte[] rowKey;
+    rowKey = Bytes.add(Bytes.toBytes(prefix.length), prefix);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionName.length), regionName);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionServerID.length), regionServerID);
+    this.key = rowKey;
+  }
+
+  public ReplicationStatusKey(byte[] key) {
+    this.key = key;
+  }
+
+  /**
+   * @return the underlying key
+   */
+  public byte[] getKey() {
+    return key;
+  }
+
+  /**
+   * Splits the keys into a {@link Splitter} which exposes the parts that comprise the key.
+   */
+  public Splitter split() {
```

Comment:
MDSKey is like that because different metadata keys consist of different numbers of parts. This key always consists of 3 parts. The entire point of this class is to put all encoding/decoding logic in one place. Forcing people to use a splitter still requires that they be aware of how it is encoded. The constructor that takes a single byte[] should get and store all the parts, and the getters can translate them to the right object.